public/script.js

```diff
@@ -71,26 +76,18 @@ async function animate(extCon, norSouth, dates, monthLoop, starting, ending) {//
 
         await map.getLayers().getArray()[0].setZIndex(10000);//add the new layer to top
         await sleep(1000);//sleep so that loading screen appears for long enough to not be confusing
-        while (map.getLayers().getArray().length < 29) {//until there are 29 layers, load in layers
-            imageURL = "https://nsidc.org/api/mapservices/NSIDC/wms?service=WMS&version=1.1.0&request=GetMap&layers=NSIDC:g02135_"+extCon+"_raster_"+monthVal+"_"+norSouth+"&styles=NSIDC:g02135_"+extCon+"_raster_basemap&bbox="+locationVal+"&format=image/png&TIME="+displayDates[i].split(" ")[1];
-            await map.addLayer(new ol.layer.Image({
-                source: new ol.source.ImageStatic({
-                    url: imageURL,
-                    projection: projection,
-                    imageExtent: extent
-                })
-            }));    
-            map.getLayers().getArray()[0].setZIndex(100);//add the new layer  
-            i++;
-        }
-
+        currentYear = Number(displayDates[0].split(" ")[1].split("-")[0]);
```

Comment:
The `moment.js` library (https://momentjs.com/docs/) might help to simplify some of these date manipulations.